sdk/src/main/scala/com/horizen/account/AccountSidechainNodeViewHolder.scala

```diff
@@ -171,6 +173,19 @@ class AccountSidechainNodeViewHolder(sidechainSettings: SidechainSettings,
       else
 
        */
+      if (tx.isInstanceOf[EthereumTransaction]) {
+        val ethTx: EthereumTransaction = tx.asInstanceOf[EthereumTransaction]
+        if (!sidechainSettings.mempool.allowUnprotectedTxs && ethTx.isLegacy && !ethTx.isEIP155) {
+          context.system.eventStream.publish(
+            FailedTransaction(
+              tx.asInstanceOf[Transaction].id,
```

Comment:
```suggestion
              tx.id
```
I think it should work, in a worse case use `ModifierId @@ tx.id`.
No sense to cast is to `Transaction` type

Reply:
tx.id works, thanks